homeassistant/components/hive/sensor.py

```diff
@@ -42,29 +43,43 @@ def device_info(self):
         return {"identifiers": {(DOMAIN, self.unique_id)}, "name": self.name}
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return FRIENDLY_NAMES.get(self.device_type)
+    def available(self):
+        """Return if sensor is available."""
+        return self.device.get("deviceData", {}).get("online", True)
 
     @property
-    def state(self):
-        """Return the state of the sensor."""
-        if self.device_type == "Hub_OnlineStatus":
-            return self.session.sensor.hub_online_status(self.node_id)
-        if self.device_type == "Hive_OutsideTemperature":
-            return self.session.weather.temperature()
+    def device_class(self):
+        """Device class of the entity."""
+        return DEVICETYPE[self.device["hiveType"]].get("type", None)
+
+    @property
+    def icon(self):
+        """Return the icon to use."""
+        return icon_for_battery_level(
+            battery_level=self.device["deviceData"]["battery"]
+        )
```

Comment:
We don't set icon when we set a device class. The battery device class will set the correct icon already in the frontend.
```suggestion
```

Reply:
Fixed in latest commit